protocol.html

```diff
@@ -717,6 +719,51 @@ <h4 property="schema:name"><span>Note</span>: Conditional Update</h4>
+
+                    <ul>
+                      <li>The patch document contains exactly one patch resource, identified by one or more of the triple patterns described above, which should all share the same <code>?patch</code> subject.</li>
+                      <li>The patch resource may optionally contain one triple of the form <code>?patch solid:matchingStrategy solid:SingleMatch</code>.</li>
+                      <li>The <code>?insertions</code> and <code>?deletions</code> formulae must not contain variables that do not occur in the <code>?conditions</code> formula.</li>
+                      <li>The <code>?insertions</code> and <code>?deletions</code> formulae must not contain blank nodes.</li>
+                      <li>The patch document does not contain any other triples.</li>
+                    </ul>
+
+                    <p><span about="" id="server-patch-n3-invalid" rel="spec:requirement" resource="#server-patch-n3-invalid"><span property="spec:statement"><span rel="spec:requirementSubject" resource="spec:Server">Servers</span> <span rel="spec:requirementLevel" resource="spec:MUST">MUST</span> respond with a <code>400</code> status code if a patch document does not satisfy all of the above constraints.</span></span></p>
+
+                    <p><span about="" id="server-patch-n3-semantics" rel="spec:requirement" resource="#server-patch-n3-semantics"><span property="spec:statement"><span rel="spec:requirementSubject" resource="spec:Server">Servers</span> <span rel="spec:requirementLevel" resource="spec:MUST">MUST</span> process a patch resource against the RDF dataset in the target document as follows, creating an empty RDF dataset if the target resource does not exist yet. If <code>?conditions</code> is non-empty, then the dataset must contain exactly one match of its contents after variable substitution; if not, the <span rel="spec:requirementSubject" resource="spec:Server">server</span> <span rel="spec:requirementLevel" resource="spec:MUST">MUST</span> respond with a <code>409</code> status code. The resulting (possibly empty) variable mapping must be propagated to the <code>?deletions</code> and <code>?insertions</code> formulae. Then, the resulting triples in <code>?deletions</code> must be removed from the RDF dataset if there is exactly one match; if not, the <span rel="spec:requirementSubject" resource="spec:Server">server</span> <span rel="spec:requirementLevel" resource="spec:MUST">MUST</span> respond with a <code>409</code> status code. Finally, the resulting triples in <code>?insertions</code> must be added to the RDF dataset, where any blank nodes must result in newly created blank nodes. The combination of the deletions and insertions then forms the new resource state of the RDF document, and the server responds with the appropriate status code.</span></span></p>
```

Comment:
This might benefit from being formatted as an ordered list after "as follows"

Reply:
I'm cautious with using explicitly numbered steps, as this gets close to the description of an imperative algorithm, which has caused trouble with specs in the past. Deferring to @csarven?

Reply:
I'm confused by there apparently being one action after "as follows:" (the list is ended by the next period; if there are meant to be multiple things that "follow", that period [and some subsequent] should instead be a semicolon).

If the (must be multiple) steps are meant to be algorithmic (though not imperatively so), an ordered list does make sense.  The implied imperative can be can be relieved by simply changing "as follows" to "roughly as follows".

If it's not so much an ordered route to conclusion, as a list of disjoint states of which only one will match the situation in play, then an unordered list would be better.

Either ordered or unordered list would be clearer than the streaming paragraph.

Reply:
Done in f9893d4156756f1d2a1388decb18bf4be527ad46